### ISSUE TITLE
A method to improve accuracy of LazyAdam

### ISSUE THREAD

**Issue by mpjlu** [tensorflow/tensorflow#23668]
LazyAdam is not convergent on NCF (MLPerf requires NCF to reach 0.635 HR in 10 epchos, but with LazyAdam HR can’t reach the goal). 

The main problem of LazyAdam is it doesn't update m and v when the gradient is 0. Actually, m and v should be updated when other weight changed.  So the convergence accuracy of LazyAdam is lower than Adam.

I have designed a method to optimize Adam on Sparse data (named SparseAdam) :　SparseAdam provides similar semantics as the original Adam algorithm, so the convergence is also close to Adam, and the TPT is about the same as LazyAdam.

**The only change of our method to LazyAdam is we compute the learning rate based the the steps skipped.**

The convergence comparing of Adam and LazyAdam on NCF (Hit Rate score):
![adam and lazyadam comparing](https://user-images.githubusercontent.com/13826327/48746725-9b11d500-ecab-11e8-8b39-947a60103890.png)

Convergence comparing of Adam, LazyAdam and SparseAdam (the proposed method) on NCF (Hit Rate score)
![adam lazyadam and sparseadam comparing](https://user-images.githubusercontent.com/13826327/48746832-08256a80-ecac-11e8-9752-3a78b79ab0e9.png)

**Comment by robieta** [comment]
Thanks. Can you open a PR putting it in `tf.contrib.opt`, and assign it to @alextp?

**Comment by mpjlu** [comment]
Thanks @robieta , I will submit PR after our internal review process.

**Comment by futurecam** [comment]
I met the same problem of slow adam in sparse, waiting your feature

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 29 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by mpjlu** [comment]
Hi all, the PR is submitted. https://github.com/tensorflow/tensorflow/pull/24788/
Thanks.

**Comment by ymodak** [comment]
Appreciate your patience. LazyAdam now falls under [tensorflow/addons](https://github.com/tensorflow/addons). The folks of [Addons - Optimizers](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#addons---optimizers) can be a good point of contact to add this feature. Thanks!
